src/scripts/main.js

```diff
@@ -1,3 +1,361 @@
+
+        table.append(...sort);
+        break;
+
+      case 'Office':
+        sort = listTable.sort((a, b) => {
+          return a.children[2].innerText.localeCompare(b.children[2].innerText);
+        });
+
+        table.append(...sort);
+        break;
+
+      case 'Age':
+        sort = listTable.sort((a, b) => {
+          return a.children[3].innerText - b.children[3].innerText;
+        });
+
+        table.append(...sort);
+        break;
+
+      case 'Salary':
+        sort = listTable.sort((a, b) => {
+          return convertNum(a.children[4].innerText)
+            - convertNum(b.children[4].innerText);
+        });
+
+        table.append(...sort);
+    }
+  }
+});
+
+table.addEventListener('click', (e) => {
+  [...document.querySelectorAll('tr')].slice(1, -1).forEach((item) => {
+    item.classList = '';
+  });
+
+  e.path[1].className = 'active';
+});
+
+const massageResultTrue = Promise.resolve('new employer created');
+const massageResultFalse = Promise.reject(Error('some wrong, write correct'));
+
+document.querySelector('button').addEventListener('click', e => {
+  const div = document.createElement('div');
+
+  div.dataset.qa = 'notification';
+
+  e.preventDefault();
+
+  const dataInput = document.querySelectorAll('input');
+  const selectSity = document.querySelector('select');
+  const optionName = document.querySelectorAll('option');
+
+  let controlForm = true;
+
+  if (dataInput[0].value.length < 4 || +dataInput[0].value) {
+    controlForm = false;
+  }
+
+  if (dataInput[1].value.length < 4 || +dataInput[1].value) {
+    controlForm = false;
+  }
+
+  if (dataInput[2].value < 18) {
+    controlForm = false;
+  }
+
+  if (dataInput[3].value < 1) {
+    controlForm = false;
+  }
+
+  if (controlForm === false) {
+    massageResultFalse
+      .catch((error) => {
+        div.classList = ('notification error');
+        div.innerText = error;
+        document.body.append(div);
+
+        setTimeout(function() {
+          div.remove();
+        }, 2000);
+      });
+
+    return;
+  }
+
+  const itemTable = document.createElement('tr');
+
+  itemTable.innerHTML = `
+    <td>${dataInput[0].value}</td>
+    <td>${dataInput[1].value}</td>
+    <td>${optionName[selectSity.selectedIndex].value}</td>
+    <td>${dataInput[2].value}</td>
+    <td>${`$` + Number(dataInput[3].value).toLocaleString(`en-ES`)}</td>
+  `;
+
+  document.querySelector('tbody').append(itemTable);
+
+  massageResultTrue
+    .then((result) => {
+      div.classList = ('notification success');
+      div.innerText = result;
+      document.body.append(div);
+
+      setTimeout(function() {
+        div.remove();
+      }, 2000);
+    });
+});
+
+const input = document.createElement('input');
+let previousText;
+
+document.querySelector('tbody').addEventListener('dblclick', e => {
+  if (e.target.tagName !== 'TD') {
+    return;
+  }
+
+  previousText = e.target.textContent;
+
+  input.className = 'cell-input';
+
+  e.target.textContent = '';
```

Comment:
```suggestion
  e.target.textContent = previousText;
```
the 2nd point:
![image](https://user-images.githubusercontent.com/86923089/205128423-471ea2b0-f8fa-42b7-bd11-c04d46b2d8d5.png)